hazelcast/src/test/java/com/hazelcast/map/MapLockTest.java

```diff
@@ -386,6 +390,28 @@ public void run() throws Exception {
         }, 30);
     }
 
+    /**
+     * See issue #4888
+     */
+    @Test
+    public void lockStoreShouldBeRemoved_whenMapIsDestroyed() {
+        HazelcastInstance hz = createHazelcastInstance();
+        IMap map = hz.getMap(randomName());
+        for (int i = 0; i < 1000; i++) {
+            map.lock(i);
```

Comment:
I guess this is my lack of understanding, but if the key is unlocked after it is locked, it the key not automatically removed? If that is the case, then there is no purpose of doing lock/unlock. So I guess my knowledge is lacking here.